Assets/BossRoom/Scripts/Client/Net/ClientGameNetPortal.cs

```diff
@@ -211,6 +218,91 @@ public static bool StartClientRelayMode(GameNetPortal portal, string roomKey, ou
             return true;
         }
 
+        public static bool StartClientUnityRelayMode(GameNetPortal portal, string joinCode, out string failMessage)
+        {
+
+            var chosenTransport  = NetworkManager.Singleton.gameObject.GetComponent<TransportPicker>().UnityRelayTransport;
+            NetworkManager.Singleton.NetworkConfig.NetworkTransport = chosenTransport;
+
+            switch (chosenTransport)
+            {
+                case UnityTransport utp:
+                    Debug.Log($"Setting Unity Relay client with join code {joinCode}");
+                    Task t = Task.Run( () => {UnityServices.InitializeAsync();});
+                    if (!t.Wait(30000))
+                    {
+                        Debug.Log("UnityServices.Initialize did not finish within 30 seconds");

```

Comment:
should we throw or early exit here if we didn't initialize properly?

Reply:
Also, is there any error messages we could display? Or any actions we could take on these errors? Let's say I get a 503 because some services are overloaded, could it be possible in a first pass to log that error and add a todo to add exponential backoff in a second pass?